### PR TITLE
CNV-8303 Make it easy to create/configure service-accounts and secrets

### PR DESCRIPTION
**Pull request by ydayagi**
…sSigned-off-by: Yaron Dayagi <ydayagi@redhat.com>Signed-off-by: Yaron Dayagi <ydayagi@redhat.com>

Signed-off-by: Yaron Dayagi <ydayagi@redhat.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by rgolangh** [comment]
@ydayagi can you cleanup the PR description?

File: README.md [hunk 1]
```diff
@@ -9,16 +9,13 @@ This repository hosts the CSI KubeVirt driver and all of its build and dependent
...
-- set RBAC rules for the service account (//TODO supply file, doc)
-- designate a namespace to deploy the VMs for the tenant cluster nodes
-- deploy tenant cluster
-- deploy files under `deploy/*`
-- edit the secret `infra-cluster-credentials` in namespace `kubevirt-csi-driver`
-    - set apiUrl: [base64 of https://<infra-cluster-api-url>]  
-    - set service-ca.crt : [base64 of <infra-cluster serivce-ca.crt>] (copy service-ca.crt value from infra-cluster-sa secret)
-    - set namespace: [base64 of <infra-cluster namespace>]
-    - set token : [base64 of <infra-cluser token>] (copy token value from infra-cluster-sa secret )
+- use `deploy/infra-cluster-service-account.yaml` to create a service account in kubevirt cluster (replace INFRA_NAMESPACE with the kubevirt cluster namepsace)
```
Comment (rgolangh) [inline README.md:12]: another option for the namespace, is to omit it from the yaml and let the creator do it on its currently logged in namespace. More tool friendly, what do you think?

**Comment by ydayagi** [comment]
> @ydayagi can you cleanup the PR description?

What do u mean by description? The 'cnv-xxx ...' title?

File: README.md (hunk 1 above)
Comment (ydayagi) [reply README.md:12]: I think that the creator will be a cluster admin in the infra. Therefore, I think that the logged in namespace might not be relvant.

Comment (ydayagi) [reply README.md:12]: I see that create command can be used with '-n' flag. I can remove the namespace form the YAML and let user specify it via command line

Comment (rgolangh) [reply README.md:12]: I'm thinking about automation mostly, but it seems that there is no way in kubectl or oc to specify a namespace flag for `create`

Comment (rgolangh) [reply README.md:12]: Actually I just tried it and one can do `oc create -f x.yaml -n mynamespace` 
but only if namespace is not defined on the yaml. I wonder if kubectl behaves the same way.

File: README.md [hunk 2]
```diff
@@ -9,16 +9,13 @@ This repository hosts the CSI KubeVirt driver and all of its build and dependent
...
+- use `deploy/secret.yaml` for creating the necessary secret in the tenant cluster
+    - set apiUrl: [base64 of infra cluster API URL. E.g. base64 of https://cluster.mydomain.co:6443]
+    - set service-ca.crt : copy value from infra service account created in previous section
```
Comment (rgolangh) [inline README.md:15]: the value is on the secret of the service account, not the service account itself

File: README.md [hunk 3]
```diff
@@ -9,16 +9,13 @@ This repository hosts the CSI KubeVirt driver and all of its build and dependent
...
+    - set namespace: copy value from infra service account created in previous section
+    - set token : copy value from infra service account created in previous section
```
Comment (rgolangh) [inline README.md:17]: same as for service-ca

File: README.md [hunk 4]
```diff
@@ -9,16 +9,13 @@ This repository hosts the CSI KubeVirt driver and all of its build and dependent
...
+- deploy files under `deploy/*` in  tenant cluster
```
Comment (rgolangh) [inline README.md:18]: the file `infra-cluster-service-account.yaml` is on that folder so `*` is not enough

Comment (ydayagi) [reply README.md]: please review

File: deploy/infra-cluster-service-account.yaml [hunk 5]
```diff
@@ -0,0 +1,33 @@
+---
```
Comment (rgolangh) [inline deploy/infra-cluster-service-account.yaml:1]: remove the leading 2 spaces for this file

Comment (ydayagi) [reply deploy/infra-cluster-service-account.yaml:1]: done

**Pull request merged**